lib/internal/crypto/hash.js

```diff
@@ -140,7 +146,74 @@ Hmac.prototype.digest = function digest(outputEncoding) {
 Hmac.prototype._flush = Hash.prototype._flush;
 Hmac.prototype._transform = Hash.prototype._transform;
 
+class Mac extends LazyTransform {
+  constructor(mac, ...args) {
+    validateString(mac, 'mac');
+
+    let nid = EVP_PKEY_HMAC;
+    let alg, key, options;
+
+    switch (mac) {
+      case 'cmac':
+        nid = EVP_PKEY_CMAC;
+        // Fall through.
+
+      case 'hmac':
+        [alg, key, options] = args;
+        validateString(alg, 'alg');
+        break;
+
+      case 'poly1305':
+        [key, options] = args;
+        nid = EVP_PKEY_POLY1305;
+        break;
+
+      case 'siphash':
+        [key, options] = args;
+        nid = EVP_PKEY_SIPHASH;
+        break;
+
+      default:
+        throw new ERR_INVALID_OPT_VALUE('mac', mac);
+    }
+
+    key = prepareSecretKey(key);
+    super(options);
+
```

Comment:
Should likely put some basic expected type checks here for `alg` at least.

Reply:
It's there a few lines up. Line 163.

Reply:
Got it... sorry, completely missed that in the first run through